### ISSUE TITLE
QUESTION: Should new posts be indexed automatically?

### ISSUE THREAD

**Issue by IIM-Arvid** [10up/ElasticPress#2244]
Hi :)

We are using elasticpress 3.5.6 on a wordpress 5.7.2 setup. Elasticsearch is running locally in version 6.3.0. At the moment 10535 documents are indexed. Health is green. So far so good.

But we have one problem for quite a long time now: If new posts get published (post-type = post and custom post-types) they are not indexed automatically. We have to trigger the reindexing process manually in the backend every day to have the new posts to show in the search/suggest results. 
So before we set up a CRON to trigger the reindex process every hour or so we wanted to clarify if this is normal behavior by the plugin or if something is wrong with our setup.
I do not rule out that there is some problem on our side, since our wordpress installation is quite big with a lot of self developed plugins, an own php framework and a huge amount of hooks we jump on.

If posts should be indexed automatically as soon as they get published then we have to find out why this does not happen. Do you know any hook in your plugin where we could hang on and test if theres something wrong? Is it even possible to trigger the reindex process with a cron?

Thank you very much for your help!

**Comment by Rahmon** [comment]
Hi @IIM-Arvid 

ElasticPress indexes automatically new posts and updated posts. When a new post is added, look if the [action_sync_on_update](https://github.com/10up/ElasticPress/blob/a01271b7a98381b89a0242c033b89eed8d8e6c2a/includes/classes/Indexable/Post/SyncManager.php#L48) is executed.

It's possible to trigger the reindex process with a cron using [WP-CLI commands provided by the plugin](https://10up.github.io/ElasticPress/tutorial-wp-cli.html).

**Comment by IIM-Arvid** [comment]
Thank you very much for the fast reply 👍 

the **action_sync_on_update** method is executed when saving/updating a post. I went through the whole method and the post object goes all the way through and is also added to the sync_queue. I looked up the bulk_indexer and so on .. no problems at all. It goes all the way down to the point at which the request is send to the ElasticSearch API. So no problems on the plugin side I guess :)

Since we are using a multi-server setup with 3 servers + loadbalancer we suspect this to be the source of the problem. 

We have now set up the reindex process with a cron using WP CLI. So it gets refreshed once a day. That should work for the moment :)

Thank you for your help.
